<?xml version="1.0"?>
<launch>

    <arg name="camera" default="D435_head_camera"/>
    <arg name="image" default="color/image_raw"/>
    <!--  raw, compressed or theora />-->
    <arg name="transport" default="compressed" /> 
    <arg name="point_cloud" value="$(arg camera)/depth/color/points" />
    
    <arg name="ref_frame" default="$(arg camera)_color_optical_frame" />
    <arg name="camera_frame" default="$(arg camera)_color_optical_frame" />
    
    <arg name="keypoint_topic" default="/detection_output_keypoint" />
    <arg name="pub_out_images" default="true" />
    <arg name="pub_out_images_topic" default="detection_output_img/compressed" />
    <arg name="show_images" default="false" />
    
    <node pkg="tpo_vision" name="laser3DTracking"
        type="laser3DTracking" output="screen" respawn="false">
        <param name="point_cloud_topic" value="/$(arg point_cloud)"/>
        
        <param name="ref_frame" value="$(arg ref_frame)"/>
        <param name="laser_spot_frame" value="laser_spot"/>
        <param name="camera_frame" value="$(arg camera_frame)"/>
        
        <param name="keypoint_topic" value="$(arg keypoint_topic)"/>
        
    </node>
    
    <!-- <node pkg="tpo_vision" name="laserSpotDetection"
        type="laserSpotDetection" output="screen" respawn="false">
        <param name="camera_image_topic" value="/$(arg camera)/$(arg image)"/>
        <param name="transport" value="$(arg transport)"/>
        <param name="camera_info_topic" value="/$(arg camera)/color/camera_info"/>
        
        <param name="pub_out_keypoint_topic" value="$(arg keypoint_topic)"/>
        <param name="pub_out_images" value="$(arg pub_out_images)"/>
        <param name="pub_out_images_topic" value="$(arg pub_out_images_topic)"/>

        <param name="show_images" value="$(arg show_images)"/>
        </node> -->
    <node pkg="tpo_vision" name="laserSpotDetectionDL"
        type="inferNodeROS.py" output="screen" respawn="false">
        <param name="camera_image_topic" value="/$(arg camera)/$(arg image)"/>
        <param name="transport" value="$(arg transport)"/>
        <param name="camera_info_topic" value="/$(arg camera)/color/camera_info"/>
        
        <param name="pub_out_keypoint_topic" value="$(arg keypoint_topic)"/>
        <param name="pub_out_images" value="$(arg pub_out_images)"/>
        <param name="pub_out_images_topic" value="$(arg pub_out_images_topic)"/>

    </node>

        
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find tpo_vision)/config/rviz/tpoLaser.rviz" >
    </node> 
  
</launch>
